php/api/class-update-fonts.php

```diff
@@ -0,0 +1,191 @@
+ * limitations under the License.
+ *
+ * @package MaterialDesign
+ */
+
+/**
+ * Class Google Fonts API .
+ *
+ * @package MaterialDesign
+ */
+
+namespace MaterialDesign\Plugin\Api;
+
+use Exception;
+use stdClass;
+use function json_decode;
+use function MaterialDesign\Plugin\get_plugin_instance;
+
+/**
+ * Class Update_Fonts
+ *
+ * @package MaterialDesign\Plugin\Api
+ */
+class Update_Fonts extends API_Base {
+
+	const TRANSIENT = 'google-fonts-json';
+
+	/**
+	 * Holds the Google Fonts API key.
+	 *
+	 * @var string|null
+	 */
+	protected $api_key;
+
+	/**
+	 * Whether to force loading from HTTP.
+	 *
+	 * @var bool
+	 */
+	public $force_http;
+
+	/**
+	 * Update_Fonts constructor.
+	 *
+	 * @param bool   $force_http Whether to force loading from HTTP. Defaults to false.
+	 * @param string $api_key    Manually pass in a Google API key. Defaults to false.
+	 */
+	public function __construct( $force_http = false, $api_key = false ) {
+		parent::__construct();
+
+		$this->force_http = $force_http;
+
+		if ( ! empty( $api_key ) ) {
+			$this->api_key = $api_key;
+		} elseif ( defined( 'GOOGLE_FONTS_API_KEY' ) && false !== GOOGLE_FONTS_API_KEY ) {
+			$this->api_key = GOOGLE_FONTS_API_KEY;
+		} else {
+			$this->api_key = null;
+		}
+
+		if ( empty( $this->api_key ) ) {
+			_material_design_error( '_material_design_no_apikey_textonly', $this->material_design_no_apikey() );
+		}
+
+		$this->endpoint        =
+			sprintf( 'https://www.googleapis.com/webfonts/v1/webfonts?key=%s&fields=items(category,variants,family)', $this->api_key );
+		$this->local_file_path = get_plugin_instance()->dir_path . '/assets/fonts/google-fonts.json';
+	}
+
+	/**
+	 * Checks if transient exists. If it does, return the contents of `assets/fonts/google-fonts.json`. If transient doesn't
+	 * exist or is expired, retrieve from Google Fonts API and store as `assets/fonts/google-fonts.json`
+	 *
+	 * @return string
+	 */
+	public function get_fonts() {
+
+		$new = null;
+
+		if ( false === get_transient( self::TRANSIENT ) || true === $this->force_http ) {
+			try {
+				$json = $this->get_http_response();
+				$new  = $this->json_to_file( $json );
+
+				set_transient( self::TRANSIENT, time(), DAY_IN_SECONDS );
+			} catch ( Exception $e ) {
+				return false;
+			}
+		} else {
+			$new = file_get_contents( get_plugin_instance()->dir_path . '/assets/fonts/google-fonts.json' );
+			$new = json_decode( $new );
+		}
+
+		$data       = new stdClass();
+		$data->data = [];
+		foreach ( $new as $name => $font ) {
+			$data->data[] = (object) [
+				'id'   => esc_attr( strtolower( $name ) ),
+				'name' => esc_html( $name ),
+			];
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Take Google Fonts JSON and turns it into our local format, then write to local file store
+	 *
+	 * @param string $json A JSON string.
+	 *
+	 * @return stdClass
+	 */
+	public function json_to_file( $json ) {
+		$data = json_decode( $json );
+
+		$fonts = new stdClass();
+		foreach ( $data->items as $font ) {
```

Comment:
Bail out if `data-items` is empty so that an empty fonts array does not get written to `google-fonts.json`